kratos/tests/cpp_tests/geometries/test_nurbs_curve_on_surface_geometry.cpp

```diff
@@ -600,15 +600,13 @@ typedef Node NodeType;
         brep_coordinates[1][0] = 0.0;
         brep_coordinates[1][1] = 3.0;
         
-        auto curve_on_surface = GenerateReferenceNurbs2dforKnotIntersections(brep_coordinates);
+        NurbsCurveOnSurfaceGeometry<3, PointerVector<Point>, PointerVector<Point>> curve_on_surface = GenerateReferenceNurbs2dforKnotIntersections(brep_coordinates);
```

Comment:
Can't you continue using the `auto` type?

```suggestion
        auto curve_on_surface = GenerateReferenceNurbs2dforKnotIntersections(brep_coordinates);
```